larndsim/lightLUT.py

```diff
@@ -0,0 +1,119 @@
+"""
+Code adapted from https://github.com/PPKoller/ArgonCubeLUTSim/blob/main/module_0/lutSim.py
+"""
+
+import sys, time, math
+import argparse
+import numpy as np
+from math import log, isnan
+from . import consts    
+
+def GetLutGeometry(lut_path): # Finds the max and minimum values of the x,y and z axis and the number of voxels in each direction
+
+    f = np.load(lut_path)
+    lut_min = np.array([f['Min'][0],f['Min'][1],f['Min'][2]])
+    lut_max = np.array([f['Max'][0],f['Max'][1],f['Max'][2]])
+    lut_ndiv = np.array([f['NDivisions'][0],f['NDivisions'][1],f['NDivisions'][2]])
+
+    return np.array([lut_min,lut_max,lut_ndiv])
+
+def GetVoxel(pos,lut_geometry): # Determines which voxel is being called based on the position of the edep. Voxels are indexed 0-2911
+
+    (lut_min,lut_max,lut_ndiv) = lut_geometry
+    vox_xyz = np.floor(pos/(lut_max-lut_min)*lut_ndiv).astype(int)+lut_ndiv/2
+    voxel = vox_xyz[2]*lut_ndiv[0]*lut_ndiv[1]+vox_xyz[1]*lut_ndiv[0]+vox_xyz[0]
+
+    return voxel
+
+
+def GetHalfDetCopy(pos): # Determines which TPC the edep takes place in
+    tpc_x = math.floor(pos[0]/(consts.ModuleDimension[0]/2.))+consts.n_mod[0]/2*consts.n_tpc[0]
+
+    return int(tpc_x)%2
+
+def LarndToLUTCoord(pos,lut_geometry): # Converts the LArND-sim coord. system to that of the LUT
+
+    # access LUT geometry
+    lut_min,lut_max,lut_ndiv = lut_geometry
+    
+    # should add another condition to be greater than whatever the thckness of the cathode is 
+    lut_pos = pos + np.array([lut_min[0],220,0])
```

Comment:
What's 220 here? we should avoid hardcoded numbers.

Reply:
This is the tpc offset, since the LUT coordinates are centered on the TPC center.  This has been fixed in https://github.com/DUNE/larnd-sim/pull/39/commits/67d16e81845f2109c01cb214f22d26f76e67266d and `consts.tpc_offsets` has been made global.  A better solution is to include this translation in the definition of the LUT itself, which will be done soon.